### DISCUSSION TITLE
`pull_request_review` event doesn't allow setting token permissions

### DISCUSSION THREAD

**Discussion by aptalca** [community/community#55940] [Actions] [answered]
### Select Topic Area

Bug

### Body

Hi,

We have a reusable workflow that manages our issue and PR tracker project by adding issues and PRs to our v2 project board and move them around based on different events.

It works great for `issues` and `pull_request_target` events, however, the `pull_request_review` event has some problems.

Even though we're setting the permissions for the github token as `issues: write`, if a PR is coming from a 3rd party fork and we approve the PR, the `pull_request_review` event triggered action has the token permissions `issues: read`. It ignores our setting in the workflow.

We also tried passing a PAT as a secret, but that's also ignored.

Is this a bug? If not, how can we get around this problem?

Our current workflow works fine when a 3rd party fork opens a PR, it works when we request a review, but it fails when we approve that PR.

Here's our calling workflow: https://github.com/linuxserver/github-workflows/blob/v1/.github/workflows/call-issue-pr-tracker.yml
Here's our reusable workflow: https://github.com/linuxserver/github-workflows/blob/v1/.github/workflows/issue-pr-tracker.yml

And here's one where we tried to pass the secret specifically: https://github.com/linuxserver/docker-mods/blob/projectsend-translations/.github/workflows/call_issue_pr_tracker.yml
And here's the reusable that accepts the passed secret: https://github.com/linuxserver/github-workflows/blob/test/.github/workflows/issue-pr-tracker.yml

Here'a failing action:
https://github.com/linuxserver/docker-mods/actions/runs/5025649402/jobs/9012899424
It shows the token permissions as `issues: read` in the set up job step, and shows the secrets as `Secret source: None`, wehereas successful actions triggered by the `pull_request_target` event for the same PR show `issues: write` and `Secret source: Actions`: https://github.com/linuxserver/docker-mods/actions/runs/5025644132/jobs/9012886960

Thanks

PS. The same workflow works fine on `pull_request_review` event if the PR is from the same repo and not a fork.

**Comment by airtower-luna** [comment, marked as answer]
That limitation is [described in the documentation](https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#workflows-in-forked-repositories-1), and it's the same as for `pull_request` events:

> With the exception of `GITHUB_TOKEN`, secrets are not passed to the runner when a workflow is triggered from a forked repository. The `GITHUB_TOKEN` has read-only permissions in pull requests from forked repositories. For more information, see "[Automatic token authentication](https://docs.github.com/en/actions/security-guides/automatic-token-authentication)."

It's a matter of security, because the workflow is used as it's at a temporary merge commit, and could've been modified maliciously.

**Comment by aptalca** [comment]
ah, so it's the same issue as `pull_request` vs `pull_request_target` but there is no `pull_request_review_target`.

Basically it is impossible to act on `pull_request_review` event to manage an issue project when the project owner approves a 3rd party PR. That's a bit silly.

**Comment by skyleaworlder** [comment]
So does this mean that there is no way to get any secrets except `GITHUB_TOKEN` as for pull request?

**Comment by slawekjaranowski** [comment]
For me also event like: `pull_request_review_target` will be helpful, I would like to update labels, milestone after approvals.
